### PR TITLE
examples: Update VISA examples to use the new session management API

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Update `nivisa_dmm_measurement` and `output_voltage_measurement` to use the new session management API.

Add more supported instrument IDs to `_visa_dmm.py`.

Update `.env.sample` and `.env.simulation`.

### Why should this Pull Request be merged?

Resolves [AB#2527580](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2527580)

### What testing has been done?

Manually tested both examples with simulation and a HP 34401A connected over GPIB.
Also tested the first example with an NI Instrument Simulator v2.0 connected over serial.

### CODE REVIEW

File: examples/nivisa_dmm_measurement/pyproject.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ PyVISA = "^1.13.0"
 PyVISA-sim = "^0.5.1"
 click = ">=7.1.2, !=8.1.4" # mypy fails with click 8.1.4: https://github.com/pallets/click/issues/2558
 grpcio = "*"
+python-decouple = ">=3.8"
```
Comment (bkeryan) [inline examples/nivisa_dmm_measurement/pyproject.toml:14]: Need to ignore missing imports for decouple

```
examples/nivisa_dmm_measurement/
  ~/work/measurementlink-python/measurementlink-python/examples/nivisa_dmm_measurement ~/work/measurementlink-python/measurementlink-python
  measurement.py:12: error: Skipping analyzing "decouple": module is installed, but missing library stubs or py.typed marker  [import-untyped]
  measurement.py:12: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
  Found 1 error in 1 file (checked 5 source files)
  Error: Process completed with exit code 1.
```

File: examples/output_voltage_measurement/measurement.py [hunk 2]
```diff
@@ -106,118 +87,100 @@ def measure(
...
-            response = measure_session.query("READ?")
-            check_instrument_error(measure_session)
-            measured_value = float(response)
-
-            source_session = None  # Don't abort after this point
+    cancellation_event = threading.Event()
+    measurement_service.context.add_cancel_callback(cancellation_event.set)
+
+    with measurement_service.context.reserve_sessions([input_pin, output_pin]) as reservation:
+        with reservation.create_nidcpower_session() as _, reservation.create_session(
```
Comment (bkeryan) [inline examples/output_voltage_measurement/measurement.py:94]: This example doesn't need the value yielded by the context manager.

@dixonjoel @pbirkhol-ni Which is less confusing?
- `with reservation.create_nidcpower_session() as _:`
- `with reservation.create_nidcpower_session():`

Comment (dixonjoel) [reply examples/output_voltage_measurement/measurement.py:94]: I prefer the latter. I think it's clear that it's just for the purpose of the session lifetime.

File: examples/output_voltage_measurement/_visa_dmm.py
Comment (dixonjoel) [inline examples/output_voltage_measurement/_visa_dmm.py:1]: Is this an identical copy of the _visa_dmm.py in nivisa_dmm_measurement? I don't think this is necessarily a problem, just wanted to understand if this an _visa_dmm_sim.yaml are the same.

File: examples/nivisa_dmm_measurement/_visa_dmm.py [hunk 3]
```diff
@@ -31,7 +31,16 @@
...
 
 # Supported NI-VISA DMM instrument IDs, both real and simulated, can be added here
-_SUPPORTED_INSTRUMENT_IDS = ["Waveform Generator Simulator", "34401"]
+_SUPPORTED_INSTRUMENT_IDS = [
+    # Keysight/Agilent/HP 34401A
+    "34401",
+    "34410",
+    "34411",
+    "L4411",
+    # NI Instrument Simulator v2.0
```
Comment (dixonjoel) [inline examples/nivisa_dmm_measurement/_visa_dmm.py:40]: Why is this one here but commented out?

**Review by dixonjoel — APPROVED** [review]

File: examples/output_voltage_measurement/_visa_dmm.py
Comment (bkeryan) [reply examples/output_voltage_measurement/_visa_dmm.py:1]: Yes, they are "vendored" and should be identical.

File: examples/nivisa_dmm_measurement/_visa_dmm.py (hunk 3 above)
Comment (bkeryan) [reply examples/nivisa_dmm_measurement/_visa_dmm.py:40]: That's a comment, not an ID. The next two lines are IDs for NI Instrument Simulator.

Comment (dixonjoel) [reply examples/nivisa_dmm_measurement/_visa_dmm.py:40]: Oh I see

File: examples/output_voltage_measurement/measurement.py (hunk 2 above)
Comment (bkeryan) [reply examples/output_voltage_measurement/measurement.py]: ok, I deleted `as _`

**Pull request merged**